src/algokit_utils/account_manager.py

```diff
@@ -0,0 +1,199 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Optional, Union, cast
+
+import algosdk
+from algosdk.account import generate_account
+from algosdk.atomic_transaction_composer import (AccountTransactionSigner,
+                                                 TransactionSigner)
+
+from .account import (get_dispenser_account, get_kmd_wallet_account,
+                      get_localnet_default_account)
+from .beta.client_manager import ClientManager
+
+
+@dataclass
+class AddressAndSigner:
+    address: str
+    signer: TransactionSigner
+
+class AccountManager:
+    """Creates and keeps track of signing accounts against sending addresses."""
+
+    def __init__(self, client_manager: ClientManager):
+        """
+        Create a new account creator.
+
+        :param client_manager: The ClientManager client to use for algod and kmd clients
+        """
+        self._client_manager = client_manager
+        self._accounts: Dict[str, TransactionSigner] = {}
```

Comment:
```suggestion
        self._accounts = dict[str, TransactionSigner]()
```

Reply:
I find this style a little easier to read, also since this library is targeted at python >= 3.10 we can use the newer typing annotation that work with builtin types